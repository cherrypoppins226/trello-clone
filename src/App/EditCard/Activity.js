import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import FormatListBulleted from "@material-ui/icons/FormatListBulleted";
import { textarea3 } from "../styled";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    "& svg": {
      color: "rgb(153, 153, 153)",
      margin: "0px 5px",
      padding: 1
    },
    "& h3": {
      fontWeight: 700,
      marginBottom: 10
    },
    "& textarea": textarea3
  },
  saveComment: {
    textTransform: "none",
    fontWeight: 700
  }
};

const Activity = ({ classes }) => (
  <div className={classes.root}>
    <div style={{ display: "flex", flexDirection: "row" }}>
      <FormatListBulleted />
      <Typography variant="subheading">Activity</Typography>
    </div>
  </div>
);

const View = withStyles(styles)(Activity);

export default View;
