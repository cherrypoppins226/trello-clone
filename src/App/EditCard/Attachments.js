import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Attachment from "@material-ui/icons/Attachment";

const styles = {
  root: {
    display: "flex",
    flexDirection: "column",
    "& svg": {
      color: "rgb(153, 153, 153)",
      margin: "0px 5px",
      padding: 1
    },
    "& h3": {
      fontWeight: 700,
      marginBottom: 10
    }
  },
  addAttachment: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    padding: 7,
    minHeight: 50,
    marginLeft: 35,
    border: "2px dashed #C4C9CC"
  }
};

const Attachments = ({ classes }) => (
  <div className={classes.root}>
    <div style={{ display: "flex", flexDirection: "row" }}>
      <Attachment />
      <Typography variant="subheading">Attachments</Typography>
    </div>
    <div className={classes.addAttachment}>
      <Typography align="center">Drag and drop or choose your files</Typography>
    </div>
  </div>
);

const View = withStyles(styles)(Attachments);

export default View;
