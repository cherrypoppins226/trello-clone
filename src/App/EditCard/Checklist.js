import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import LinearProgress from "@material-ui/core/LinearProgress";
import { fileAbsolute } from "paths.macro";

import { moduleName } from "../utils";

const styles = {
  checklist: {
    display: "flex",
    alignItems: "center"
  },
  progress: {
    marginLeft: 10,
    width: "100%",
    background: "#D6DADC",
    height: 8,
    borderRadius: 4
  },
  addItem: {
    marginTop: 10,
    textTransform: "none"
  }
};

const View = ({ classes }) => (
  <React.Fragment>
    <div className={classes.checklist}>
      <Typography variant="caption">0%</Typography>
      <LinearProgress
        className={classes.progress}
        variant="determinate"
        value={0}
      />
    </div>
    <Typography
      component={Button}
      className={classes.addItem}
      size="small"
      align="left"
    >
      Add an item...
    </Typography>
  </React.Fragment>
);

const Styled = withStyles(styles)(View);

Styled.displayName = moduleName(fileAbsolute);

export default Styled;
