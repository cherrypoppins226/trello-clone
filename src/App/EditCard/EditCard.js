import React from "react";
import { findDOMNode } from "react-dom";
import { withStyles } from "@material-ui/core/styles";
import { fileAbsolute } from "paths.macro";

import { moduleName } from "../utils";
import Header from "./Header";
import Sidebar from "./Sidebar";
import Content from "./Content";
import { cardType } from "../CardsList/Card";
import * as labels from "../labels";

const styles = {
  root: {
    position: "absolute",
    top: 60,
    width: 750,
    backgroundColor: "rgb(237, 239, 240)",
    borderRadius: 2,
    outline: "none"
  },
  header: {
    margin: "20px 20px 30px 15px"
  },
  content: {
    float: "left",
    width: "75%",
    padding: "0px 10px"
  },
  sidebar: {
    float: "right",
    width: "25%"
  }
};

const centerInPage = node => {
  node.style.left = `${(window.innerWidth - node.offsetWidth) / 2}px`;
};

class View extends React.Component {
  componentDidMount() {
    this.node = findDOMNode(this);
    centerInPage(this.node);
  }

  render() {
    const { classes, card, ...props } = this.props;
    return (
      <div
        aria-describedby={labels.editCardDescription.id}
        className={classes.root}
        {...props}
      >
        <Header className={classes.header} text={card.title} />
        <Content className={classes.content} />
        <Sidebar className={classes.sidebar} />
      </div>
    );
  }
}

const Styled = withStyles(styles)(View);

Styled.displayName = moduleName(fileAbsolute);

Styled.propTypes = {
  card: cardType.isRequired
};

export default Styled;
