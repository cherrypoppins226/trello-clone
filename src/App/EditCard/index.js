import React from "react";
import { findDOMNode } from "react-dom";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Title from "./Title";
import * as Labels from "../labels";

const styles = {
  root: {
    position: "fixed",
    top: 60,
    height: 700,
    width: 750,
    backgroundColor: "rgb(237, 239, 240)",
    borderRadius: 2,
    outline: "none"
  },
  title: {
    margin: "8px 32px 8px 46px",
    fontWeight: 700
  }
};

const centerInPage = node => {
  node.style.left = `${(window.innerWidth - node.offsetWidth) / 2}px`;
};

class EditCard extends React.Component {
  componentDidMount() {
    this.node = findDOMNode(this);
    centerInPage(this.node);
  }

  render() {
    const { classes, title, ...props } = this.props;
    return (
      <div
        aria-describedby={Labels.editCardDescription.id}
        className={classes.root}
        {...props}
      >
        <Title className={classes.title} text={title} />
      </div>
    );
  }
}

const View = withStyles(styles)(EditCard);

View.propTypes = {
  title: PropTypes.string.isRequired
};

export default View;
