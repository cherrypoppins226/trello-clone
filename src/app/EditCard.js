import React from "react";
import { findDOMNode } from "react-dom";
import { withStyles } from "@material-ui/core/styles";
import { fileAbsolute } from "paths.macro";
import { compose } from "recompose";
import { inject } from "mobx-react";

import { makeFixtures, labelId, moduleName } from "../utils";
import Header from "./editCard/Header";
import Sidebar from "./editCard/Sidebar";
import Content from "./editCard/Content";

const modulePath = moduleName(fileAbsolute);

export const labels = {
  description: {
    id: labelId(modulePath, "description"),
    text: "Edit card: Change title, descriptions, labels, attachments, and more"
  }
};

const styles = {
  root: {
    width: 750,
    backgroundColor: "rgb(237, 239, 240)",
    borderRadius: 2,
    outline: "none"
  },
  flex: {
    display: "flex"
  },
  content: {
    width: "100%",
    padding: "0px 10px"
  },
  sidebar: {
    flexBasis: 220
  }
};

// TODO: Shouldn't be a concern here
const centerInPage = node => {
  node.style.left = `${(window.innerWidth - node.offsetWidth) / 2}px`;
};

class EditCard extends React.Component {
  componentDidMount() {
    this.node = findDOMNode(this);
    centerInPage(this.node);
  }

  render() {
    const { classes, appState, ...rest } = this.props;
    return (
      <div
        aria-describedby={labels.description.id}
        className={classes.root}
        tabIndex={-1}
        {...rest}
      >
        <Header
          className={classes.header}
          cardId={appState.cardBeingEdited.id}
          cardTitle={appState.cardBeingEdited.title}
        />
        <div className={classes.flex}>
          <Content className={classes.content} />
          <Sidebar className={classes.sidebar} />
        </div>
      </div>
    );
  }
}

const Component = compose(inject("appState"), withStyles(styles))(EditCard);

export const fixtures = makeFixtures(modulePath, Component, {
  default: {
    props: {},
    stores: {
      appState: {
        cardBeingEdited: { id: 1, title: "Ut sunt qui amet." },
        finishCardEdit: () => {}
      }
    }
  }
});

export default Component;
