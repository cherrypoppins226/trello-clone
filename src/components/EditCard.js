import React from "react";
import path from "path";
import PropTypes from "prop-types";
import { findDOMNode } from "react-dom";
import { withStyles } from "material-ui/styles";
import Grid from "material-ui/Grid";
import Modal from "material-ui/Modal";
import Typography from "material-ui/Typography";
import Button from "material-ui/Button";
import green from "material-ui/colors/green";
import grey from "material-ui/colors/grey";
import Archive from "@material-ui/icons/Archive";
import Label from "@material-ui/icons/Label";
import Timer from "@material-ui/icons/Timer";
import LibraryBooks from "@material-ui/icons/LibraryBooks";
import ArrowForward from "@material-ui/icons/ArrowForward";
import Person from "@material-ui/icons/Person";
import { cardDescription } from "./CardsListCard";
import TextArea from "./TextArea";

const styles = {
  container: {
    outline: "none",
    pointerEvents: "none"
  },
  editCardDescription: {
    "& textarea": {
      pointerEvents: "all",
      padding: 4,
      paddingLeft: 8,
      // Reusable
      border: 0,
      outline: "none",
      resize: "none",
      borderRadius: 2
    },
    "& button": {
      pointerEvents: "all",
      color: grey[50],
      backgroundColor: green[500],
      textTransform: "none",
      fontWeight: 700,
      marginTop: 15,
      "&:hover": {
        backgroundColor: green[600]
      }
    }
  },
  editCardButtons: {
    "& button": {
      pointerEvents: "all",
      background: "rgba(0, 0, 0, .6)",
      color: grey[50],
      fontWeight: 600,
      textTransform: "none",
      transition: "transform 85ms ease-in",
      "&:hover": {
        background: "rgba(0, 0, 0, .8)",
        transform: "translateX(5px)",
        transition: "transform 85ms ease-in"
      },
      "& svg": {
        padding: 1,
        marginRight: 5,
        width: "0.8em",
        height: "0.8em"
      }
    }
  }
};

export const EDIT_CARD_DESCRIPTION = "edit-card-description";

const View = ({ classes, onClose, card = null, container = null }) => {
  const modal = React.createRef();
  const textarea = React.createRef();

  const modalDidMount = () => {
    const cardCoordinates = card.getBoundingClientRect();
    const modalNode = findDOMNode(modal.current);
    modalNode.style.top = `${cardCoordinates.top}px`;
    modalNode.style.left = `${cardCoordinates.left}px`;
    const textareaNode = findDOMNode(textarea.current);
    textareaNode.style.width = `${cardCoordinates.width}px`;
    textareaNode.style.height = `${cardCoordinates.height + 50}px`;
  };

  return (
    <Modal
      ref={modal}
      aria-describedby={EDIT_CARD_DESCRIPTION}
      open={Boolean(card)}
      container={container || card}
      onClose={onClose}
      onRendered={modalDidMount}
    >
      <Grid container wrap="nowrap" spacing={8} className={classes.container}>
        <div id={EDIT_CARD_DESCRIPTION} style={{ display: "none" }}>
          Edit card: change title, members, archive etc.
        </div>
        <Grid item className={classes.editCardDescription}>
          <Typography
            ref={textarea}
            component={TextArea}
            value={card ? cardDescription(card) : ""}
          />
          <Button variant="raised"> Save </Button>
        </Grid>
        <Grid item className={classes.editCardButtons}>
          <Grid container spacing={8} wrap="nowrap" direction="column">
            {[
              [Label, "Edit Labels"],
              [Person, "Change Members"],
              [ArrowForward, "Move"],
              [LibraryBooks, "Copy"],
              [Timer, "Change Due Date"],
              [Archive, "Archive"]
            ].map(([Icon, text], idx) => (
              <Grid item style={{ paddingBottom: 1 }} key={idx}>
                <Button size="small" variant="flat">
                  <Icon />
                  {text}
                </Button>
              </Grid>
            ))}
          </Grid>
        </Grid>
      </Grid>
    </Modal>
  );
};

View.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.object.isRequired,
  card: PropTypes.object,
  container: PropTypes.object,
  onClose: PropTypes.func.isRequired
};

View.displayName = path.basename(__filename, path.extname(__filename));

export default withStyles(styles)(View);
