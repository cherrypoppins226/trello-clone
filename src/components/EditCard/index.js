import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import * as Labels from "../labels";

const styles = {};

const EditCard = ({ classes, card = null, ...extra }) => {
  return <div aria-describedby={Labels.editCardDescription.id} {...extra} />;
};

const View = withStyles(styles)(EditCard);

View.propTypes = {
  card: PropTypes.object
};

export default View;
