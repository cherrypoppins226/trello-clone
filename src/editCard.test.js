import React from "react";
import { render } from "./__tests__/helpers.js";
import EditCard from "./editCard";

it("initial render", () => {
  const { container } = render(
    <EditCard card={null} parent={{}} onClose={jest.fn()} />
  );
  expect(container).toMatchSnapshot();
});
