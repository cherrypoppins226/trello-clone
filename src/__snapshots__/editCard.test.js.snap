// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial render 1`] = `
<div
  style="overflow: hidden; padding-right: 0px;"
>
  <div
    aria-hidden="true"
  />
  <div
    aria-describedby="edit-card-description"
    class="MuiModal-root"
    style="top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      class="MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-wrap-xs-nowrap View-container-60"
      role="document"
      tabindex="-1"
    >
      <div
        id="edit-card-description"
        style="display: none;"
      >
        Edit card: change title, members, archive etc.
      </div>
      <div
        class="MuiGrid-item View-editCardDescription-61"
      >
        <textarea
          class="MuiTypography-root MuiTypography-body1"
          spellcheck="false"
          style="width: 0px; height: 50px;"
        >
          Title...
        </textarea>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-raised"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             Save 
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-item View-editCardButtons-62"
      >
        <div
          class="MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap"
        >
          <div
            class="MuiGrid-item"
            style="padding-bottom: 1px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M17.63 5.84C17.27 5.33 16.67 5 16 5L5 5.01C3.9 5.01 3 5.9 3 7v10c0 1.1.9 1.99 2 1.99L16 19c.67 0 1.27-.33 1.63-.84L22 12l-4.37-6.16z"
                    />
                  </g>
                </svg>
                Edit Labels
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-item"
            style="padding-bottom: 1px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                    />
                  </g>
                </svg>
                Change Members
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-item"
            style="padding-bottom: 1px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </g>
                </svg>
                Move
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-item"
            style="padding-bottom: 1px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
                    />
                  </g>
                </svg>
                Copy
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-item"
            style="padding-bottom: 1px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M15 1H9v2h6V1zm-4 13h2V8h-2v6zm8.03-6.61l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42C16.07 4.74 14.12 4 12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9 9-4.03 9-9c0-2.12-.74-4.07-1.97-5.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                    />
                  </g>
                </svg>
                Change Due Date
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-item"
            style="padding-bottom: 1px;"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M20.54 5.23l-1.39-1.68C18.88 3.21 18.47 3 18 3H6c-.47 0-.88.21-1.16.55L3.46 5.23C3.17 5.57 3 6.02 3 6.5V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V6.5c0-.48-.17-.93-.46-1.27zM12 17.5L6.5 12H10v-2h4v2h3.5L12 17.5zM5.12 5l.81-1h12l.94 1H5.12z"
                    />
                  </g>
                </svg>
                Archive
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
